src/System/extract_system_rep.rs

```diff
@@ -149,6 +204,72 @@ impl SystemRecipe {
             },
         }
     }
+
+    /// Gets the height of the `SystemRecipe` tree
+    fn height(&self) -> usize {
+        match self {
+            SystemRecipe::Composition(l, r)
+            | SystemRecipe::Conjunction(l, r)
+            | SystemRecipe::Quotient(l, r, _) => max(l.height(), r.height()),
+            SystemRecipe::Component(_) => 1,
+        }
+    }
```

Comment:
This function will always return 1 for any `SystemRecipe`
I think `max(l.height(), r.height())` should be `1 + max(l.height(), r.height())`

Reply:
Nice spotted, thank you! Is fixed now